llvm/unittests/Target/AIE/AIEScheduleDAGMITest.cpp

```diff
@@ -29,9 +29,10 @@ namespace {
 class DummyAIEHazardRecognizer : public AIEHazardRecognizer {
 public:
   DummyAIEHazardRecognizer(const ScheduleDAG *DAG)
-      : AIEHazardRecognizer(nullptr, nullptr, DAG, /*ScoreboardDepth=*/0) {}
+      : AIEHazardRecognizer(nullptr, nullptr, *(new AIEAlternateDescriptors()),
+                            DAG, /*ScoreboardDepth=*/0) {}
 
-  ~DummyAIEHazardRecognizer() override {}
+  ~DummyAIEHazardRecognizer() override { delete &getSelectedAltDescs(); }
```

Comment:
Can you do the same as in PRASAieHazardRecognizer to avoid dynamic allocation?

Reply:
Thank you, have updated accordingly!